src/hooks/mining/useMiningUiStateMachine.ts

```diff
@@ -0,0 +1,60 @@
+import { useAppStateStore } from '@app/store/appStateStore';
+import { useAppStatusStore } from '@app/store/useAppStatusStore';
+import { useCPUStatusStore } from '@app/store/useCPUStatusStore';
+import { useGPUStatusStore } from '@app/store/useGPUStatusStore';
+import { useMiningStore } from '@app/store/useMiningStore';
+import { setAnimationState } from '@app/visuals';
+import { useEffect } from 'react';
+import { useMiningControls } from './useMiningControls';
+
+export const useUiMiningStateMachine = () => {
+    const { handleStart } = useMiningControls();
+
+    const isMiningInitiated = useMiningStore((s) => s.miningInitiated);
+    const setIsChangingMode = useMiningStore((s) => s.setIsChangingMode);
+    const isChangingMode = useMiningStore((s) => s.isChangingMode);
+
+    const isAutoMiningEnabled = useAppStatusStore((s) => s.auto_mining);
```

Comment:
can we use `useShallow` when grabbing items from `useAppStatusStore` especially, since it's updated every second